### PR TITLE
initial commit for coinbase handler

### PR DESCRIPTION
**Pull request by bustedware**
## Description

Pulls candle data from [coinbase api](https://docs.cloud.coinbase.com/exchange/reference/exchangerestapi_getproductcandles)
Feature request complete for [issue](https://github.com/mindsdb/mindsdb/issues/8151)

## Type of change

- [x] ⚡ New feature (non-breaking change which adds functionality)

## Verification Process

To ensure the changes are working as expected:

 - [x]   Test Location: NA / test locally
 - [x]   Verification Steps: Outline the steps or queries needed to validate the change. Include any data, configurations, or actions required to reproduce or see the new functionality.

Walk through the handlers readme, similar to binance handler

## Checklist:

- [x] My code follows the style guidelines(PEP 8) of MindsDB.
- [x] I have appropriately commented on my code, especially in complex areas.
- [x] Necessary documentation updates are either made or tracked in issues.
- [x] Relevant unit and integration tests are updated or added.

### CODE REVIEW

File: mindsdb/integrations/handlers/coinbase_handler/coinbase_handler.py [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+from mindsdb.integrations.libs.api_handler import APIHandler
+from mindsdb.integrations.libs.const import HANDLER_CONNECTION_ARG_TYPE as ARG_TYPE
+from mindsdb.integrations.libs.response import (
+    HandlerStatusResponse as StatusResponse,
+    HandlerResponse as Response,
+)
+
+from mindsdb_sql import parse_sql
+
+_BASE_COINBASE_US_URL = 'https://api.exchange.coinbase.com'
```
[the remark this answers is not in the document]
Comment (bustedware) [reply mindsdb/integrations/handlers/coinbase_handler/coinbase_handler.py:22]: thanks. ill look into this at another time.

File: mindsdb/integrations/handlers/coinbase_handler/coinbase_handler.py [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+
+        Returns StatusResponse indicating whether or not the handler is connected.
+        """
+        response = StatusResponse(True)
+        self.is_connected = response.success
+        return response
+
+    # symbol: BTC-USD
+    # granularity 60, 300, 900, 3600, 21600, 86400
+    def get_coinbase_candle(self, symbol, granularity):
```
[the remark this answers is not in the document]
Comment (bustedware) [reply mindsdb/integrations/handlers/coinbase_handler/coinbase_handler.py]: complete

File: mindsdb/integrations/handlers/coinbase_handler/coinbase_handler.py [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+        current_time = datetime.datetime.now()
+        start_time = current_time - datetime.timedelta(seconds=granularity)
+        start_time_iso = start_time.isoformat().split(".")[0] + "-04:00"
+        path = "/products/" + symbol + "/candles?granularity=" + str(granularity) + "&start=" + start_time_iso
+        headers = self.generateApiHeaders("GET", path)
+        url = _BASE_COINBASE_US_URL + path
+        response = requests.get(url, headers=headers)
+        candles = response.json()
+        for candle in candles:
+            dt = datetime.datetime.fromtimestamp(candle[0], pytz.timezone('US/Eastern')).isoformat()
```
[the remark this answers is not in the document]
Comment (bustedware) [reply mindsdb/integrations/handlers/coinbase_handler/coinbase_handler.py]: I set `pytz.timezone('US/Eastern')` to `None` and I see the same result which means it must be using system time and is now the intended result.

File: mindsdb/integrations/handlers/coinbase_handler/coinbase_handler.py [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+    def _get_candle(self, params: Dict = None) -> pd.DataFrame:
+        """Gets aggregate trade data for a symbol based on given parameters
+
+        Returns results as a pandas DataFrame.
+
+        Args:
+            params (Dict): Trade data params (symbol, interval)
+        """
+        if 'symbol' not in params:
+            raise ValueError('Missing "symbol" param to fetch trade data for. (BTC-USD)')
```
[the remark this answers is not in the document]
Comment (bustedware) [reply mindsdb/integrations/handlers/coinbase_handler/coinbase_handler.py]: fixed

File: mindsdb/integrations/handlers/coinbase_handler/coinbase_handler.py [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+    HandlerStatusResponse as StatusResponse,
+    HandlerResponse as Response,
+)
+
+from mindsdb_sql import parse_sql
+
+_BASE_COINBASE_US_URL = 'https://api.exchange.coinbase.com'
+
+
+class CoinBaseHandler(APIHandler):
```
[the remark this answers is not in the document]
Comment (bustedware) [reply mindsdb/integrations/handlers/coinbase_handler/coinbase_handler.py:25]: added tests

File: mindsdb/integrations/handlers/coinbase_handler/coinbase_handler.py [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+            raise ValueError('Missing "interval" param (60, 300, 900, 3600, 21600, 86400).')
+
+        candle = self.get_coinbase_candle(params['symbol'], int(params['interval']))
+        return candle
+
+    def native_query(self, query: str = None) -> Response:
+        ast = parse_sql(query, dialect='mindsdb')
+        return self.query(ast)
+
+    def generateApiHeaders(self, method, path):
```
[the remark this answers is not in the document]
Comment (bustedware) [reply mindsdb/integrations/handlers/coinbase_handler/coinbase_handler.py]: this has been completed

**Review by ZoranPandovski — APPROVED** [review]

**Pull request merged**